### PR TITLE
Support for Apple Pay on web

### PR DESCRIPTION
**Pull request by korycins**
- add secrettext field to give a possibility to upload multiline secret data.
- add endpoint paymentInitialize which can be used to activate session on the payment provider side
- add plugin method to initialize the payment
- add logic to activate apple pay session
- add logic to check apple certificate provided by user

### CODE REVIEW

File: saleor/graphql/payment/types.py [hunk 1]
```diff
@@ -152,3 +152,14 @@ def resolve_credit_card(root: models.Payment, _info):
         if not any(data.values()):
             return None
         return CreditCard(**data)
+
+
+class PaymentInitialized(graphene.ObjectType):
+    class Meta:
+        description = "Represents a initialized data by payment gateway."
+
+    gateway = graphene.String(description="Payment gateway.", required=True)
```
Comment (fowczarek) [inline saleor/graphql/payment/types.py:161]: Maybe?
```suggestion
    gateway = graphene.String(description="Id of payment gateway.", required=True)
```

File: saleor/payment/gateways/adyen/utils.py [hunk 2]
```diff
@@ -350,3 +354,70 @@ def get_payment_method_info(
         brand=brand, type="card" if payment_method == "scheme" else payment_method,
     )
     return payment_method_info
+
+
+def validate_payment_data_for_apple_pay(
```
Comment (fowczarek) [inline saleor/payment/gateways/adyen/utils.py:359]: Maybe split Adyen utils into small files per provider?
For example `saleor/payment/gateways/adyen/utils/apple_pay.py`

**Review by fowczarek — APPROVED** [review]

File: saleor/payment/gateways/adyen/utils.py (hunk 2 above)
Comment (korycins) [reply saleor/payment/gateways/adyen/utils.py:359]: sounds good

File: saleor/graphql/payment/types.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/graphql/payment/types.py:161]: "ID of a payment gateway" 🚔

File: saleor/payment/gateways/adyen/plugin.py [hunk 3]
```diff
@@ -168,6 +181,17 @@ class AdyenGatewayPlugin(BasePlugin):
             ),
             "label": "Enable native 3D Secure",
         },
+        "apple-pay-cert": {
+            "type": ConfigurationTypeField.SECRET_TEXT,
+            "help_text": (
+                "Follow the Adyen docs related to activating the apple pay for the "
```
Comment (maarcingebala) [inline saleor/payment/gateways/adyen/plugin.py:187]: "Apple Pay" is a proper name so please fix all occurrences to use proper capitalization.

File: saleor/plugins/base_plugin.py [hunk 4]
```diff
@@ -33,12 +34,14 @@ class ConfigurationTypeField:
...
     BOOLEAN = "Boolean"
     SECRET = "Secret"
+    SECRET_TEXT = "SecretText"
     PASSWORD = "Password"
     CHOICES = [
         (STRING, "Field is a String"),
         (BOOLEAN, "Field is a Boolean"),
         (SECRET, "Field is a Secret"),
         (PASSWORD, "Field is a Password"),
+        (SECRET_TEXT, "Field is a Secret multiline"),
```
Comment (maarcingebala) [inline saleor/plugins/base_plugin.py:44]: Maybe `SECRET_MULTILINE` would be a better name? Just wondering, `SECRET` and `SECRET_TEXT` sounds the same to me.

File: saleor/graphql/payment/types.py [hunk 5]
```diff
@@ -152,3 +152,14 @@ def resolve_credit_card(root: models.Payment, _info):
...
+    name = graphene.String(description="Payment gateway name.", required=True)
+    data = graphene.JSONString(
+        description="Initiliazed data by gateway.", required=False
```
Comment (maarcingebala) [inline saleor/graphql/payment/types.py:164]: Can we add a word or two about what this data is needed for? From the descriptions in this mutation I have no idea what this mutation is needed for.

File: saleor/graphql/payment/types.py [hunk 6]
```diff
@@ -152,3 +152,14 @@ def resolve_credit_card(root: models.Payment, _info):
         if not any(data.values()):
             return None
         return CreditCard(**data)
+
+
+class PaymentInitialized(graphene.ObjectType):
+    class Meta:
+        description = "Represents a initialized data by payment gateway."
```
Comment (maarcingebala) [inline saleor/graphql/payment/types.py:159]: "Represents an initialized payment." ?

**Review by IKarbowiak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**